### PR TITLE
Ability to specify the interface to listen on for the broker listeners

### PR DESCRIPTION
**Pull request by ldom**
# Description

Per host, in case there are multiple interfaces and you need to listen on only one of them, you can also specify which interface to listen on, per listener.
On each listener, you need to add an 'ip' value. This will adjust the 'listeners' section per host.

Inventory example is provided.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

We have tested it at the customer site.

**Test Configuration**:

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] Any variable changes have been validated to be backwards compatible

### CODE REVIEW

File: roles/variables/vars/main.yml [hunk 1]
```diff
@@ -132,7 +132,7 @@ kafka_broker_properties:
       zookeeper.connect: "{{ groups['zookeeper'] | default(['localhost']) | confluent.platform.resolve_hostnames(hostvars) | join(':' + zookeeper_client_port|string + ',') }}:{{zookeeper_client_port}}{{zookeeper_chroot}}"
       log.dirs: "{{ kafka_broker.datadir | join(',') }}"
       listener.security.protocol.map: "{% for listener in kafka_broker_listeners|dict2items %}{% if loop.index > 1%},{% endif %}{{ listener['value']['name'] }}:{{ listener['value'] | confluent.platform.kafka_protocol_defaults(ssl_enabled, sasl_protocol)}}{% endfor %}"
-      listeners: "{% for listener in kafka_broker_listeners|dict2items %}{% if loop.index > 1%},{% endif %}{{ listener['value']['name'] }}://:{{ listener['value']['port'] }}{% endfor %}"
+      listeners: "{% for listener in kafka_broker_listeners|dict2items %}{% if loop.index > 1%},{% endif %}{{ listener['value']['name'] }}://{{ custom_listeners['properties'][listener['key']]['ip'] | default('') }}:{{ listener['value']['port'] }}{% endfor %}"
```
Comment (domenicbove) [inline roles/variables/vars/main.yml:135]:
```suggestion
      listeners: "{% for listener in kafka_broker_listeners|dict2items %}{% if loop.index > 1%},{% endif %}{{ listener['value']['name'] }}://{{ listener['value']['ip'] | default('') }}:{{ listener['value']['port'] }}{% endfor %}"
```

`kafka_broker_custom_listeners` gets merged into `kafka_broker_listeners`, so i think `ip` will end up there if you set it. Please test this change out

Also please update the target branch to a .x branch

**Review by domenicbove — CHANGES_REQUESTED** [review]
good start, some edits!

File: roles/variables/vars/main.yml (hunk 1 above)
Comment (ldom) [reply roles/variables/vars/main.yml]: Just tested it with your suggested change and it's all good. Thanks

File: hosts_example.yml [hunk 2]
```diff
@@ -315,6 +315,21 @@ kafka_broker:
...
+      # kafka_broker_custom_listeners:
+      #   broker:
+      #     ip: 123.123.123.123
+      #     hostname: 'broker1.something.local'
+      #   internal:
+      #     ip: 123.123.123.123
+      #     hostname: 'broker1.something.local'
+      #   client_listener:
+      #     ip: 123.123.123.123
+      #     hostname: 'broker1.something.local'
```
Comment (domenicbove) [inline hosts_example.yml:331]:
```suggestion
      #     hostname: broker1.something.local
      #   internal:
      #     ip: 123.123.123.123
      #     hostname: broker1.something.local
      #   client_listener:
      #     ip: 123.123.123.123
      #     hostname: broker1.something.local
```

these quotes arent necessary right?

**Review by domenicbove — CHANGES_REQUESTED** [review]
please change target branch to 7.0.x, but LGTM!

File: hosts_example.yml (hunk 2 above)
Comment (ldom) [reply hosts_example.yml]: no, you're right... removed

**Comment by ldom** [comment]
> please change target branch to 7.0.x, but LGTM!

done

**Review by domenicbove — APPROVED** [review]
Nice! We should get this added to our official docs as well @nsharma-git

**Review by domenicbove — APPROVED** [review]

**Review by nsharma-git — APPROVED** [review]

**Pull request merged**